### ISSUE TITLE
Add dark theme

### ISSUE THREAD

**Issue by xmha97** [yakovlevegor/DroidRec#16]
Please add dark theme to this app. (Like picture below)

![Screenshot_2022-11-04-00-56-55-243_com.yakovlevegor.DroidRec.jpg](https://user-images.githubusercontent.com/23485114/199837577-d0b4f394-130f-4889-a5e7-2a85148fbd49.jpg)

**Comment by drogga** [comment]
@yakovlevegor
+1, but the Status, Navigation and the action bar(s) should also be colored the same, in order to look seamless (unlike the status bar from the OP's example, where it kind of looks ugly, since it's a concept), a pure black "oled"\"amoled" themed UI layout, instead of a dark grey\gray one also works for me, as long as it's not white or whitish.

BTW, I noticed that the quick tile buton is always colored (active) on Samsung Galaxy S8+ with Android 9 Pie, instead of being greyed-out when the app isn't recording and only colored when it does, like it should be and like all of the other quick tile buttons, being a system ones, or from third-party apps.

**Comment by drogga** [comment]
@yakovlevegor
+1, but the Status, Navigation and the action bar(s) should also be colored the same, in order to look seamless (unlike the status bar from the OP's example, where it kind of looks ugly, since it's a concept), a pure black "oled"|"amoled" themed UI layout, instead of a dark grey\gray one also works for me, as long as it's not white or whitish.

Maybe you can also center everything in order to be better and more easily reachable on tall screens...?

BTW, I noticed that the quick tile buton is always colored (active) on Samsung Galaxy S8+ with Android 9 Pie, instead of being greyed-out when the app isn't recording and only colored when it does, like it should be and like all of the other quick tile buttons, being a system ones, or from third-party apps.

**Comment by xmha97** [comment]
> The dark theme will be added, but would be controlled from system settings.

Good.

**Comment by yakovlevegor** [comment]
@drogga, as for the quick tile, I'll just make it always inactive, as it only redirects to an activity that starts the recording, and doesn't start it by itself.

**Comment by yakovlevegor** [comment]
@drogga, nevermind, I've added state monitoring for the tile icon anyway.

**Comment by yakovlevegor** [comment]
@drogga, elements are centered on the screen now.

**Comment by drogga** [comment]
@yakovlevegor, Thanks for everything, but the app's minSDKversion is 21=5.0, Android 9 or 8 on Samsung have a Dark Theme (Night Mode) system toggle, OK, the app now reads this and switches its theme accordingly, but on older OS version, like 5, 6, 7 and 8 (non-Samsung) there's no toggle there, there are third-party apps that can force this even on 4.3 Jelly Bean, but there's no way from the app to change \ switch the theme, which means - always white on below A.9, also what if someone still wants for some strange reason to have both with different themes ?, like the system in dark, but the app in white, or the opposite - the app in dark, but the system in\with white theme...

Also, I apologize in advance for not creating a different \ separate issue tickets for each thing, but I have some other observations, ideas and suggestions, listed below:
The NavBar is still in white;

By default, the notification when the recording starts is with highest priority, like it pops-up (heads-up), makes sound and vibration, I've manually changed it, because I can, but some devices and older Android version do not have such control over them, maybe changing the priority of it in the code to normal (maybe not silent or minimized) will be good.

Is there a reason on why you haven't added storage permissions, but handling the storage write access in such way, giving exclusive external storage access ?, IDK how and if this even works on alder OS vers...

Maybe because of the video's meta tag, the way is encoded, or something else, but for some reason, unlike with other videos, the playback speed rate control in the Samsung's build-in system video player do not show up on screen record from this all, which is strange.

Can it be made to live toggle the audio capturing while recording, instead of having to stop and start over ?, I mean if I start without checkmark of any of the 2 options (I have only a "mic" one on A.9), but I change my mind and don't want to start over, I can still put a checkmark mark on any of the audio recording options, but it wouldn't take effect until I stop and start again, the opposite too - if I start with audio, but don't want to have (record) one for the whole video..., of course, I can edit it out, but the app being able to do this audio on\off thing, while recording would be awesome.

**Comment by drogga** [comment]
@yakovlevegor, Thanks for everything, but the app's minSDKversion is 21=5.0, Android 9 or 8 on Samsung have a Dark Theme (Night Mode) system toggle, OK, the app now reads this and switches its theme accordingly, but on older OS version, like 5, 6, 7 and 8 (non-Samsung) there's no toggle there, there are third-party apps that can force this even on 4.3 Jelly Bean, but there's no way from the app to change \ switch the theme, which means - always white on below A.9, also what if someone still wants for some strange reason to have both with different themes ?, like the system in dark, but the app in white, or the opposite - the app in dark, but the system in\with white theme...

Also, I apologize in advance for not creating a different \ separate issue tickets for each thing, but I have some other observations, ideas and suggestions, listed below:
The NavBar is still in white;

By default, the notification when the recording starts is with highest priority, like it pops-up (heads-up), makes sound and vibration, I've manually changed it, because I can, but some devices and older Android version do not have such control over them, maybe changing the priority of it in the code to normal (maybe not silent or minimized) will be good.

Is there a reason on why you haven't added storage permissions, but handling the storage write access in such way, giving exclusive external storage access ?, IDK how and if this even works on older OS vers...

Can it be made to live toggle the audio capturing while recording, instead of having to stop and start over ?, I mean if I start without checkmark of any of the 2 options (I have only a "mic" one on A.9), but I change my mind and don't want to start over, I can still put a checkmark mark on any of the audio recording options, but it wouldn't take effect until I stop and start again, the opposite too - if I start with audio, but don't want to have (record) one for the whole video..., of course, I can edit it out, but the app being able to do this audio on\off thing, while recording would be awesome.

**Comment by yakovlevegor** [comment]
@drogga,

>OK, the app now reads this and switches its theme accordingly, but on older OS version, like 5, 6, 7 and 8 (non-Samsung) there's no toggle there

The dark theme should be supported only on devices that have dark mode option. I don't think it's a good idea to add unnecessary customizations. Needless to say that to change notification style on older systems would require hundreds of megabytes of libraries bundled with the application. I just keep the building process as simple as possible, thus avoiding adding Android Jetpack trash to the bundle.

>By default, the notification when the recording starts is with highest priority, like it pops-up (heads-up), makes sound and vibration

The recording start is announced, that's normal.

>Can it be made to live toggle the audio capturing while recording, instead of having to stop and start over ?

The recording starts over to avoid problems with screen resolution.

**Comment by yakovlevegor** [comment]
@drogga,

>Is there a reason on why you haven't added storage permissions

Yes, because the app uses DocumentsProvider API, which is more convenient for choosing a folder to store recordings.

**Comment by yakovlevegor** [comment]
@drogga, as for colored notifications, Android supports them natively only since version 8. I could add the manual dark theme switch, but notifications on Android versions below 8 won't be colored.

**Comment by yakovlevegor** [comment]
@drogga, ah, nevermind. The background color doesn't work as expected. It's not possible to change it without altering system settings. I could add theme toggle for main activity, but that's all I can do.

**Comment by drogga** [comment]
I haven't suggested coloring the notifications at all, just the NavBar (Navigation Bar, the thing on bottom of the screen, with the 3 buttons for back, home and recents).

Yes, a simple dark or not theme switch with a checkmark would be enough, it shouldn't require any fancy coding, overcomplicate the app, or making it way larger in size, by using libs and dependencies, I don't want that too.

But, if you can at least fix the auto-rotation (orientation) bug, would be awesome, so it doesn't crash when I rotate the device\screen or I launch a game or a app which is landscape only and it rotates the screen, this is a must.

Some other thing to maybe consider in future (which will most definitely increase the size, but it will also make it feature-rich and not so simple\basic:

Native FPS, instead of 30 (even on a 120 Hz) screen, most devices are at least with 60 as minimum, as default or with a picker option.

Rezolution picker.

BitRate picker.

Custom file naming.

Floating controls for pause\record and stop, should be as a single movable button, which when tapped will expand vertically and show the other button, so it wont get in the way when interacting with the screen.

Show touches option (which can also be among the other floating buttons).

"ScreenCam" (v2.0.3) has those and can also record internal audio and the mic at the same time, without root\magisk and on lower than A.10 and is ~2 MB

BTW, thers's no need to mention me, especially in the commits, I'm subscribed, I will see and get notified when you comment ;)

**Comment by yakovlevegor** [comment]
@drogga,

>But, if you can at least fix the auto-rotation (orientation) bug, would be awesome, so it doesn't crash when I rotate the device\screen or I launch a game or a app which is landscape only and it rotates the screen, this is a must.

It doesn't **crash** when the screen rotates, this is a feature to avoid screen resolution problems.

**Comment by drogga** [comment]
No, problem, I did just throw some ideas, you decide if it's worth implementing \ adding ;)

Yes, the recording can also be stopped from the notification, but if you have other ones there, that shouldn't be recorder in case you send the video or attach it somewhere, in that case you have to either stop it form the app itself, or edit it to cut that part.

**Comment by yakovlevegor** [comment]
@drogga, well, if there is a crash, could you please specify your device specs? I'll run it in an emulator to check it out.

**Comment by yakovlevegor** [comment]
@drogga, oh, right.

**Comment by drogga** [comment]
More specifically: SM-G955F

**Comment by yakovlevegor** [comment]
@drogga, tried it in an emulator, when I rotate screen on Android 9 recording doesn't start anew. Definitely a bug.

**Comment by yakovlevegor** [comment]
Should be fixed now, using different API listener to detect screen orientation change. Works on Android emulator.

**Comment by drogga** [comment]
Will you build it and make a release so I can test it please ?, or maybe setup a git actions workflow to do it automatically...

**Comment by yakovlevegor** [comment]
If you're up to testing, sure. Here's a debug build.
[app-build-debug.zip](https://github.com/yakovlevegor/DroidRec/files/10031054/app-build-debug.zip)

**Comment by drogga** [comment]
Nothing change, it's the same.

**Comment by yakovlevegor** [comment]
Didn't detect any errors on Android 9. Your model has resolution `1440 x 2960`, that could be abnormal. I just created #18, mind if we move there for that bit?

**Comment by drogga** [comment]
I don't mind ;)

**Comment by yakovlevegor** [comment]
As for native FPS, it only can be added for Android 10+, because settings recording's FPS manually with `MediaRecorder` moves it to time lapse mode, and disables audio recording completely.

**Comment by yakovlevegor** [comment]
I can set changing video's BitRate according to FPS, but setting the FPS manually for < Android 10 I can't do.

**Comment by yakovlevegor** [comment]
Oh, wait, I can, through `setFrameRate`. Apologies.

**Comment by yakovlevegor** [comment]
On some devices the native FPS is not supported though.

**Comment by yakovlevegor** [comment]
Moreover, the set FPS value isn't respected by `MediaRecorder` at all.

**Comment by yakovlevegor** [comment]
I tried recording 60 FPS on an emulator with 60 native framerate, and on my phone with 90 native framerate, the video is recorded in 30 FPS regardless of the preset value. And on Android 10+ it crashes, because Android's H264 codec doesn't support recording in more than 30 FPS by itself.

**Comment by yakovlevegor** [comment]
`setVideoFrameRate` sets maximum framerate, not a constant one, I get it now. But the problem still persists for Android 10+, it doesn't record when FPS isn't 30.

**Comment by yakovlevegor** [comment]
Looks like the error is in the codec picking function. Need to change it to a more smarter codec choosing.

**Comment by yakovlevegor** [comment]
Yes, with a different codec the error vanishes. Need to add a check for supported framerate when picking a codec.

**Comment by drogga** [comment]
Thanks to this: https://github.com/yrom/ScreenRecorder I know what you're going through with the codecs, It's a mess. that's how I'll put it.

**Comment by yakovlevegor** [comment]
Just came to a conclusion that codecs give placeholders instead of real FPS capabilities. Instead I need to test them all to determine one that doesn't crash.

**Comment by yakovlevegor** [comment]
About BitRate, maybe instead I should add like a quality scale with three states, from Normal to Low.

**Comment by drogga** [comment]
Or maybe use Average BitRate (ABR) or Variable BitRate (VBR), instead of Constant BitRate (CBR), the same can probably be done for the Frame Rate, I've tested some other recorders recently and only a few of them had "Auto" options for the 2, which should mean that instead of locking it to a preset, like say 4 MBps or 120/native FPS in order to look good, it can probably self-adjust, which can also result in lower file size, because it won't try to push\keep those frames or bitrate even when is not necessary, like on slow movement\objects\frames, or when tge image is still\static. « just an idea.

**Comment by yakovlevegor** [comment]
I'm already adding an advance settings panel to cover that up.

**Comment by drogga** [comment]
OK 😉, but if it doesn't work, then forget about the bitrate option picker, add a theme one and create a new release...

**Comment by yakovlevegor** [comment]
An app can't set a bitrate constant, it completely depends on the codecs that are chosen.

**Comment by yakovlevegor** [comment]
A panel has been added. The floating controls are next.

**Comment by yakovlevegor** [comment]
Sure, I should be naming artifacts accordingly.

**Comment by yakovlevegor** [comment]
The panel will be added for Android >= 8. I get crashes when trying to create an overlay on older versions.

**Comment by yakovlevegor** [comment]
Version [1.8](https://github.com/yakovlevegor/DroidRec/releases/tag/v1.8) is out. This issue is now closed.

**Comment by drogga** [comment]
When I go to the settings, the StatusBar becomes from black to light grey.

It straight-up crashes when I rotate to landscape, even if the app is not in focus, but in the background, and not doing anything, also when I try to directly launch it while in landscape, it doesn't work at all like this, only in portrait does, but again - no LogCat, so I can't provide any. Tho it works while it records, which is strange.

Recording quality - the slider is a bit janky, when I press "cancel", it still remembers the position, but the text is not always correct.

No hint or min\max limitation in\for the "Recording fps", I cam put 0 > unlimited number, it will accept it.

"Recording bitrate" - same thing, I can put from 0 to whatever many numbers I like in the input field there, it will accept it, no hint as of min.\max. value and the unit is unknown, is it in kbps, mbps or what, I should guess.

When putting a checkmark on "Enable floating controls" and I haven't granted the overlay permission, it should ask then, not when I press the "Start Recording" button on the main window, also the floating buttons can be smaller, because they are a bit large and their (custom) position is not being remembered on the screen for portrait or landscape when rotating, during the recording, because only when it records it doesn't crash while rotating.

**Comment by yakovlevegor** [comment]
>It straight-up crashes when I rotate to landscape

>it still remembers the position

>No hint of min\max limitation in\for the "Recording fps"

>"Recording bitrate" - same thing

Thx, working on it.

**Comment by drogga** [comment]
It will be better if the floating control buttons are vertical, or at least in portrait orientation, and like now (horizontal) in landscape, tho it will be much better and easier for me to use\reach them if they are vertical in both orientations.

You might get complains of users on why their records are missing from the gallery, because most users don't use file managers (lol), while the video files are actually in the rec. dir.\folder, but for some reason the media scanner didn't get triggered, it happens to me, I wrote it many-many comments ago.

**Comment by drogga** [comment]
It will be better if the floating control buttons are vertical, or at least in portrait orientation, and like now (horizontal) in landscape, tho it will be much better and easier for me to use\reach them if they are vertical in both orientations.

You might get complains by users on why their records are missing from the gallery, because most users don't use file managers (lol), while the video files are actually in the rec. dir.\folder, but for some reason the media scanner didn't get triggered, it happens to me, I wrote it many-many comments ago. IDK why it doesn't index only some.

**Comment by drogga** [comment]
It will be better if the floating control buttons are vertical, or at least in portrait orientation, and like now (horizontal) in landscape, tho it will be much better and easier for me to use\reach them if they are vertical in both orientations.

You might get complains by users on why their records are missing from the gallery, because most users don't use file managers (lol), while the video files are actually in the rec. dir.\folder, but for some reason the media scanner didn't get triggered, it happens to me, I wrote it many-many comments ago. IDK why it doesn't index only some.

**Comment by yakovlevegor** [comment]
I don't think I get it about the buttons. The panel is collapsible.

**Comment by drogga** [comment]
Basically 90°? rotated.

**Comment by drogga** [comment]
No always collapses and expands, it's buggy and I broke it, at the end only the handle of it shows up :D, force stopping the app fixed it, until I c\e it multiple times again (because it's janky)

**Comment by yakovlevegor** [comment]
>You might get complains by users on why their records are missing from the gallery

I think people should choose their recordings directory, like in NewPipe, for example.

**Comment by yakovlevegor** [comment]
>no LogCat

Scoop doesn't catch any exceptions. To collect logs through LogCat you need adb.

**Comment by yakovlevegor** [comment]
That does it for another release. Thanks for testing. If you encounter any more errors, please open a new issue. This is spammed upon so horribly.

**Comment by yakovlevegor** [comment]
Version [1.9](https://github.com/yakovlevegor/DroidRec/releases/tag/v1.9) is out.

**Comment by drogga** [comment]
> I think people should choose their recordings directory, like in NewPipe, for example.

I did and this happens, IDK who or what should trigger the MediaScanner, DroidRec, SAF, or else, and tell it to reScan and index the videos from the recording folder, being a custom or the default one for every single recording, being manually started or automatically by DroidRec when rotating during recording.


> Scoop doesn't catch any exceptions. To collect logs through LogCat you need adb.

Yes, Scoop captures\catches only the crashes of a particular app, not the exceptions like those, the "LogCat Extreme" app that I used before grabs all the system logs and you can also grab only the one of a particular app, like how Scoop does.
I don't have the ability currently to use ADB (I obviously know how to use it, since I activated Scoop months ago), that's why I only rely on Scoop at the moment.


> This is spammed upon so horribly.

I agree. If you want, I can start deleting my own comments ?, those that are not related to the GUI\layout, or you can even delete the issue if you like and maybe enable the discussions tab...

> Version 1.9 is out.

Will test it, thanks.

**Comment by yakovlevegor** [comment]
>and tell it to reScan and index the videos from the recording folder

No need, DocumentsProvider creates and saves documents only through file descriptors, avoiding standard file creation apis.

>If you want, I can start deleting my own comments

No, my point was that this issue is off-topic, since theme settings were already added.

**Comment by drogga** [comment]
You can rename the issue ticket if you want, but but then next things I write are very much also GUI related and are where they should be - here, so not off-topic at all:

The input fields are not much better, still no hints whatever and no explanation on the bitrate value used and all the related things I wrote earlier before.

The floating control overflow always starts centered based on the NavBar, its custom position is not being remembered.

I don't exactly get how those buttons can be collapsed \ expanded in the panel's handle, single tapping it worked once, double-tapping it whenever it wants, it's hit or miss..., please explain or fix it. I like that they are now vertical in portrait.

**Comment by yakovlevegor** [comment]
>The input fields are not much better

I've added a placeholder and zero input detection, just like you asked.

>The floating control overflow always starts centered

That's how it should be, on each new recording.

>I don't exactly get how those buttons can be collapsed

By tapping on the handle without moving it.

**Comment by yakovlevegor** [comment]
>The input fields are not much better

Oh, I get it now. I shouldn't update them on the go.

**Comment by yakovlevegor** [comment]
>it's hit or miss

The function checks if coordinates before and after motion are the same. I'm adding a motion range which will be ignored, so it will be easier to press the panel handle.

**Comment by yakovlevegor** [comment]
Version [2.0](https://github.com/yakovlevegor/DroidRec/releases/tag/2.0) is out.

**Comment by yakovlevegor** [comment]
Version [2.1](https://github.com/yakovlevegor/DroidRec/releases/tag/2.1) is out.

**Comment by drogga** [comment]
Good Job !
Can you please maybe set some transparency on the 2 floating buttons (the vector .png images), like by 50% ?, I don't think there's a need for opacity slider or something.

The bitrate field is now set by default to "71928000", it will still accept ridiculous\insane values, like "9999999999999999" and more and as low as 1, along with the FPS field (the setting's title & description of it should be with all caps, like "FPS", not "fps" - super minor thing), but most importantly - most users won't know how much is this (=71,928 Mbps » Super \ unnecessarily High, the max. of "ScreenCam" is 48 Mbps), that's why I thought, next to the "Set custom bitrate" description, to write something like: "`{The value is in Kbps, 1000 Kbps = 1 Mbps, default value = 71928000}`" or 1024.

"Floating controls size" mine is now set on "Small", but I thing it can go even smaller (tiny) and the time count will still be visible - just saying. ;)

Out of 3 recordings, only 1 was indexed my the media scanner and was shown in the gallery app, which is a system on Samsung devices and shows Images and Videos, other times it shows 2 of 3 there, it's very chaotic, with many recordings, at least 1 is always not shown (indexed), same test with ScreenCam and all are indexed and shown in the gallery, that's why I'm mentioning it for at least 3rd time, not a much of an issue for me personally, but I can see it being a huge no-no for others, because as I mentioned - most people, mainly only rely on the gallery apps for images and videos, instead of a file managers, like I do, plus when you go to upload \ attach a rec. somewhere (to sent) not all will not show up in the galleries.

**Comment by yakovlevegor** [comment]
>The bitrate field is now set by default to "71928000"

It discards itself each time the setting gets deactivated. The value is an automatically calculated one, which screen recorder uses by default.

>most users won't know how much is this

Most users should use the automatic preset based on their device specs and custom quality settings, not reinvent the wheel with bitrate settings. It's an advanced feature.

>I think it can go even smaller

It won't be clickable. Microscopic size is just not useful.

**Comment by drogga** [comment]
Apologies for the many edits, you can see where they are, what's edited and reRead only that.
I wanted to be perfectly clear and explain those as detailed as possible, sorry.

**Comment by drogga** [comment]
> Most users should use the automatic preset based on their device specs and custom quality settings, not reinvent the wheel with bitrate settings. It's an advanced feature.

You are correct, that's also true, NVM then, pls forget that I suggested it.


> It won't be clickable. Microscopic size is just not useful.

The "Small" ones are with the size of my pinky finger on my big scrren (6.2"), so they will be easy to manage \ press even if their size is reduced by 30%, tho my hands a regular male size, not huge sausages for fingers :D, you might say: define "regular", but OK, you are the developer, you decide what's best.

**Comment by yakovlevegor** [comment]
I've made a debug build with a function to query the file path with `MediaScannerConnection` after each recording.
[app-build-scanmedia.zip](https://github.com/yakovlevegor/DroidRec/files/10096788/app-build-scanmedia.zip)

**Comment by drogga** [comment]
6 of 6 indexed, nice job.
It felt like it crashed after the first recording when I pressed the Stop Recording button, it didn't show anything as a dialog or something, but it worked flawlessly after that, so it's probably something expected. You can push the release.

**Comment by yakovlevegor** [comment]
Thanks for testing. Just gonna think about a `Tiny` panel and bitrate boundaries, and renaming "fps" to "FPS".

**Comment by drogga** [comment]
👍

**Comment by drogga** [comment]
You might also add a screenshot with the dark theme in the readme, or replace that one, ideally would be a animated GIF with the 2 changing, that way it would attract more users - I think, also most of the sites I visit are with a dark theme, either support it or Bromite forces it for me, so that white one is a bit too much on the eyes when I visit the repo. ;)

**Comment by yakovlevegor** [comment]
Just noticed the integer overflow in bitrate and framerate pickers. I just got to limit characters to 10, as max Integer value is (2^31)-1, otherwise Java crashes.

**Comment by drogga** [comment]
Ok. the bitrate is with 8 currently for me and I don't think that any FPS, Codec\Encoder or Rez. will need more than that, it's hight enough already.

**Comment by yakovlevegor** [comment]
Max fps value will be 300, max bitrate 250 mbit/s.

**Comment by drogga** [comment]
Agree.

**Comment by yakovlevegor** [comment]
I think that's all. Here's the latest debug build. I need to wait for Chinese translation anyway.
[app-build-debug.zip](https://github.com/yakovlevegor/DroidRec/files/10097184/app-build-debug.zip)

**Comment by yakovlevegor** [comment]
Version [2.2](https://github.com/yakovlevegor/DroidRec/releases/tag/v2.2) is out.

**Comment by drogga** [comment]
v2.3:
The first time (after being closed) it still splits the recordings when rotating (which I liked, so I suggest you to put a option whether to split it, or to continue recording, which makes it not optimal when viewing), but after the second time the continuous recording works, so it's a bug.
Also the first time after updating it, the Open With from the notification didn't work, it still threw this toast msg.: "Error: Recording failed", which didn't, they were actually 2 recs., because I rotated, couldn't reproduce by trying 3 times after that.

A wrong intent is probably being used, because the Open With dialog that pops-up from the notification doesn't show all the options I usually have for the media files (Audio .m4a - Mono, Single Ch., my device can record in stereo, it has multiple mics., 48 kHz, has frame rate for some reason, but it doesn't bit rate, maybe add a option for like 128 kbps and mono\stereo ?), like for the Audio, it shows 2 apps "Termux" & "Intent Intercept", a file manager shows also "MediaInfo" & an actual player that can play this, in the face of "Samsung player" (system, pops-up) & "Samsung Music" (the same, it should've been the full player, but it's not, nvm, not your fault), Video - "Video Player", Google "Photos", "Intent Intercept" & "Termux", from a file manager, I also have the "MediaInfo" option there too for the .mp4 to open with. If the screen locks, the Open With dialog that is being called from the rec. notification is gone, along with it. I haven't tried the Share or Delete options from there.

**Comment by drogga** [comment]
This is already a mess, that's why I wrote that here, I didn't want the latest issue ticket to become the same, tho if you insist, I can copy-paste it there and delete that comment above from here...

**Comment by yakovlevegor** [comment]
No, that's fine. The notification issue is Android 9-specific.

**Comment by yakovlevegor** [comment]
Here's the pre-built with audio sample rate set to fixed value.
[app-build-fixedsamplerate.zip](https://github.com/yakovlevegor/DroidRec/files/10121910/app-build-fixedsamplerate.zip)

**Comment by yakovlevegor** [comment]
And I've decided to add that option after all. But mind that you're losing half your quality on that one.
[app-build-norotate-option.zip](https://github.com/yakovlevegor/DroidRec/files/10122421/app-build-norotate-option.zip)

**Comment by yakovlevegor** [comment]
And I've added an option to disable new recording notification on rotate, to remove the transitions.
[app-build-dontnotifyonrotate.zip](https://github.com/yakovlevegor/DroidRec/files/10122667/app-build-dontnotifyonrotate.zip)

**Comment by yakovlevegor** [comment]
Here, the notification is fixed.
[app-build-dontnotifyonrotate-fixed.zip](https://github.com/yakovlevegor/DroidRec/files/10122944/app-build-dontnotifyonrotate-fixed.zip)

**Comment by drogga** [comment]
I've started with "app-build-norotate-option.zip", updating over the official v2.3, it crashed at first, but not crashes after that and I did a lot of test, 10+, I thought by looking at the commits, that this will be the default behavior (continuous rec.), that why I requested the option, so it can be as before, which I prefer (the app to split them when rotating), so I won't be keeping the option enabled, but it's nice to have it for everyone else, that has a diff. opinion.

The sample rate is now lowered to 44.1 kHz, OK, no need to be 48, tho I still don't know the audio's bitrate, since the "MediaInfo" doesn't show it for those recordings for some reason, but it does for other audio files, and it's still in Mono (single channel), even tho my and most of the devices, since 2013 can record in Stereo too.

The audio files do not play when opening them in "Samsung Music", but as I wrote before, picking the same form the "open with" form a file manager does play them, the notification'a send action, intent or whatever is probably not correct for the audio recordings, now also "MT Manager" shows as a option in the "open with" prompt dialog, which is strange, but at least the "MediaInfo" shows too, too bad that for now only the audio player doesn't get called properly.

When locking the device\screen, the "Open With" should stay on the screen, the notification too, up until an action is chosen, but they still don't after unlocking, both are dismissed.

"app-build-dontnotifyonrotate-fixed.zip" - No notification while recording at all, only after I press the stop button or rotate there's a "done" or "finished" one.

**Comment by drogga** [comment]
The whole time the "Don't notify on rotate" oprion was Off, I haven't tried how it will behave if it's On.

**Comment by yakovlevegor** [comment]
Fixed the audio not recording/sharing properly, I chose a raw aac without a mpeg container, which caused encoding errors.

**Comment by drogga** [comment]
OK.
I just tried with "Don't notify on rotate" Enabled - it does what it describes, I wasn't bothered by it, since mine are set to normal priority without a sound for this app, but it's still a nice one to have ;)

**Comment by yakovlevegor** [comment]
When I install 2.3 and then install "app-build-dontnotifyonrotate-fixed.zip" notifications show all according to settings.

**Comment by yakovlevegor** [comment]
Here's a build with audio format fixed.
[app-build-fixedformat.zip](https://github.com/yakovlevegor/DroidRec/files/10123740/app-build-fixedformat.zip)

**Comment by yakovlevegor** [comment]
I get a straight up crash when trying to record audio on Android 9.

**Comment by drogga** [comment]
In the meantime, I did a clean install of the "FixedFormat" .APK and the notification with the controls not show up while recording, everything else seems as I described it.

**Comment by yakovlevegor** [comment]
`DocumentsContract` just crashes when trying to rename the file from ".mp4" to ".m4a".

**Comment by drogga** [comment]
Might help [Audio form the notif.》Open With》Intent Intercept]: 
intent:///storage/emulated/0/DCIM/Screenshots/AudioRecording_20221130_162837.m4a#Intent;scheme=file;type=audio/mp4;launchFlags=0x13000000;end 
------------ 
ACTION: android.intent.action.VIEW 
DATA: file:///storage/emulated/0/DCIM/Screenshots/AudioRecording_20221130_162837.m4a 
MIME: audio/mp4 
URI: intent:///storage/emulated/0/DCIM/Screenshots/AudioRecording_20221130_162837.m4a#Intent;scheme=file;type=audio/mp4;launchFlags=0x13000000;end 
FLAGS: 
FLAG_RECEIVER_FOREGROUND 
FLAG_ACTIVITY_FORWARD_RESULT 
FLAG_ACTIVITY_PREVIOUS_IS_TOP 

------------ 
MATCHING ACTIVITIES: 
MT Manager (bin.mt.plus - bin.mt.plus.OpenFileActivity) 
Samsung Music (com.sec.android.app.music - com.samsung.android.app.music.activity.SoundPlayerActivity) 
Termux (com.termux - com.termux.filepicker.TermuxFileReceiverActivity) 
MediaInfo (net.mediaarea.mediainfo - net.mediaarea.mediainfo.ReportListActivity) 
Sound player (com.samsung.android.app.soundpicker - com.samsung.android.app.musiclibrary.ui.player.soundplayer.SoundPlayerActivity)

ReSending the intent from there, shows this toast: 
file:///storage/emulated/0/DCIM/Screenshots/AudioRecording_20221130_162837.m4a exposed beyond app through ClipData.Item.getUri()

**Comment by drogga** [comment]
The stupid markdown f-ed the formatting, I hope that you can understand it like it is...

**Comment by yakovlevegor** [comment]
Gonna look at the intent. Just fixed the DocumentsContract, the exceptions were firing here and there, giving tons of false positives, had to disable them.

**Comment by yakovlevegor** [comment]
The error happens because of policy restrictions. I've set them loose in previous build (worked for me on Android 9). Made a new build that refreshes them every new recording.
[app-build-policyupdate.zip](https://github.com/yakovlevegor/DroidRec/files/10124355/app-build-policyupdate.zip)

**Comment by yakovlevegor** [comment]
Or this one, I've set it explicitly to "lax".
[app-build-policylax.zip](https://github.com/yakovlevegor/DroidRec/files/10124486/app-build-policylax.zip)

**Comment by xmha97** [comment]
Why not create separate issues for different discussions? In this issue, I wanted the dark theme to be added, which was done.

**Comment by yakovlevegor** [comment]
Yes, @drogga. I suggest you create a new one.

**Comment by drogga** [comment]
Well, you can't blame, I asked...

**Comment by yakovlevegor** [comment]
Ok, ok. Do it already. 😄
